configs/components/openssl-3.0.rb

```diff
@@ -128,13 +128,18 @@
     # 'no-bf', pgcrypto is requires this cipher in postgres for puppetdb
     # 'no-cast', pgcrypto is requires this cipher in postgres for puppetdb
     'no-rc5',
-    # 'no-md4', puppet infra uses the agent's runtime and runs WinRM tasks using NTLM, so it needs DES & MD4
     'no-mdc2',
     # 'no-rmd160', this is causing failures with pxp, remove once pxp-agent does not need it
-    'no-whirlpool',
-    'no-legacy'
+    'no-whirlpool'
   ]
 
+  if settings[:use_legacy_openssl_algos]
+    pkg.apply_patch 'resources/patches/openssl/openssl-3-activate-legacy-algos.patch'
+  else
+    configure_flags << 'no-legacy' << 'no-md4' << 'no-des'
```

Comment:
This ticket (https://tickets.puppetlabs.com/browse/PE-36078) enumerates the changes need, and it says "no-des" should be included when legacy algos are _not_ enabled, but.... it wasn't included before.

I added it thinking it might have been an oversight in the original Puppet 8 work, but would love to know if this is overkill and/or actively not wanted in the puppet-runtime. /cc @joshcooper

Reply:
It like you read my mind :grin: We had plans on adding `no-des` in https://tickets.puppetlabs.com/browse/PA-5587, but put it on hold when PE started failing due to `no-md4`. But now that you've detangled that, I'm :+1: to this PR